Kernel/VM/MemoryManager.cpp

```diff
@@ -908,17 +902,11 @@ NonnullRefPtrVector<PhysicalPage> MemoryManager::allocate_contiguous_supervisor_
     VERIFY(!(size % PAGE_SIZE));
     ScopedSpinLock lock(s_mm_lock);
     size_t count = ceil_div(size, static_cast<size_t>(PAGE_SIZE));
-    NonnullRefPtrVector<PhysicalPage> physical_pages;
-
-    for (auto& region : m_super_physical_regions) {
-        physical_pages = region.take_contiguous_free_pages(count);
-        if (!physical_pages.is_empty())
-            continue;
-    }
+    auto physical_pages = m_super_physical_region->take_contiguous_free_pages(count);
 
     if (physical_pages.is_empty()) {
-        if (m_super_physical_regions.is_empty()) {
-            dmesgln("MM: no super physical regions available (?)");
+        if (!m_super_physical_region) {
```

Comment:
Minor note and hold-over from previous version, can we remove the "(?)" at the end of the message? It confused me when I hit it and made me think something *super* unusual was happening. If there is more to be learned, maybe it can be added to the message?

Reply:
It is indeed super unusual to hit it (and it is not the message you hit, you hit the one a couple lines after :) as we VERIFY its non-null immediately after creation, but in this case the error is not even possible (unlike the before version), as we would have crashed earlier on a dereference. So I'll just remove it completely.